crates/subspace-service/src/lib.rs

```diff
@@ -924,6 +953,11 @@ where
                 "worker",
                 Some("sync-from-dsn"),
                 Box::pin(async move {
+                    // Run snap-sync before DSN-sync.
+                    if config.sync == ChainSyncMode::Snap {
+                        let _ = snap_sync_task.await;
```

Comment:
This will ensure that we get compiler warning if/when we decide to return something like `Result` out of it instead of current `()`:
```suggestion
                        snap_sync_task.await;
```